### ISSUE TITLE
Unable to filter on website id with product repository

### ISSUE THREAD

**Issue by joost-florijn-kega** [magento/magento2#7396]
### Preconditions
1. PHP 5.6
2. Magento 2 dev branch

### Steps to reproduce
1. Create model with the StoreManager, SearchCriteriaBuilder and ProductRepository passed by dependency injection.
2. Filter product repository on website_id
```
$website = $this->storeManager->getWebsite();
$searchCriteria = $this->searchCriteriaBuilder
            ->addFilter('website_id', $website->getId())
            ->create();

$result = $this->productRepository->getList($searchCriteria); 
```

### Expected result
1. List of products filtered by website.

### Actual result
1. Exception was thrown when adding the fetching the result.
Exception:
```
exception 'Magento\Framework\Exception\LocalizedException' with message 'Invalid attribute name: website_id' in vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php:1370
Stack trace:
#0 vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php(1460): Magento\Eav\Model\Entity\Collection\AbstractCollection->_addAttributeJoin('website_id', 'inner')
#1 vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php(362): Magento\Eav\Model\Entity\Collection\AbstractCollection->_getAttributeConditionSql('website_id', Array, 'inner')
#2 vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php(1521): Magento\Eav\Model\Entity\Collection\AbstractCollection->addAttributeToFilter(Array, NULL, 'inner')
#3 var/generation/Magento/Catalog/Model/ResourceModel/Product/Collection/Interceptor.php(519): Magento\Catalog\Model\ResourceModel\Product\Collection->addAttributeToFilter(Array, NULL, 'inner')
#4 vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php(393): Magento\Catalog\Model\ResourceModel\Product\Collection\Interceptor->addAttributeToFilter(Array, NULL)
#5 var/generation/Magento/Catalog/Model/ResourceModel/Product/Collection/Interceptor.php(870): Magento\Eav\Model\Entity\Collection\AbstractCollection->addFieldToFilter(Array, NULL)
#6 vendor/magento/module-catalog/Model/ProductRepository.php(718): Magento\Catalog\Model\ResourceModel\Product\Collection\Interceptor->addFieldToFilter(Array)
#7 vendor/magento/module-catalog/Model/ProductRepository.php(669): Magento\Catalog\Model\ProductRepository->addFilterGroupToCollection(Object(Magento\Framework\Api\Search\FilterGroup), Object(Magento\Catalog\Model\ResourceModel\Product\Collection\Interceptor))
#8 var/generation/Magento/Catalog/Model/ProductRepository/Interceptor.php(89): Magento\Catalog\Model\ProductRepository->getList(Object(Magento\Framework\Api\SearchCriteria))
#9 Magento\Catalog\Model\ProductRepository\Interceptor->getList(Object(Magento\Framework\Api\SearchCriteria))
```

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-66740

**Comment by magento-engcom-team** [comment]
@joost-florijn-kega, thank you for your report.
The issue is already fixed in 2.2.0
